### ISSUE TITLE
`vvi_from_sf()` uses ptr slot of `"SpatRaster"` which no loger exists (?)

### ISSUE THREAD

**Issue by wlangera** [STBrinkmann/GVI#11]
Hi

In the `vvi_from_sf()` function, there is a check to see if the `x` and `y` resolution are the same (see #1).
This is done based on the `ptr` slot of the `"SpatRaster"` object.
However, this slot does not exist for me. Perhaps because something changed in the **terra** package or one of its dependencies.
Yet, I cannot find any info on this ...

Either way, the slot is now called `cpp` or perhaps a safer way is to use the `res()` function.

I provide a reproducible example and also my current workaround:

``` r
library(terra)
#> Warning: package 'terra' was built under R version 4.3.2
#> terra 1.7.65

# Load example raster file
gs_path <- file.path("Manchester_LULC_green.tif")

# Downlaod GreenSpace mask from Zenodo. For details of see:
# https://doi.org/10.5281/zenodo.5061257
download.file(url = paste0("https://zenodo.org/record/5061257/files/"
                           , "GreaterManchester_GreenSpace_5m.tif"),
              destfile = gs_path, mode = "wb")

gs <- rast(gs_path)

# x and y resolution must be equal
# See https://github.com/STBrinkmann/GVI/issues/1

## code from GVI::vvi_from_sf
try(gs@ptr$res[1] != gs@ptr$res[2])
#> Error in try(gs@ptr$res[1] != gs@ptr$res[2]) : 
#>   no slot of name "ptr" for this object of class "SpatRaster"

## working code
try(gs@cpp$res[1] != gs@cpp$res[2])
#> [1] TRUE

try(res(gs)[1] != res(gs)[2])
#> [1] TRUE


# Current workaround
## Create a new class definition with the additional slot
setClass(
  "SpatRasterWithPtr",
  contains = "SpatRaster",
  slots = c(ptr = "ANY")
)

## Create a constructor function for the new class
SpatRasterWithPtr <- function(...) {
  obj <- new("SpatRasterWithPtr", ...)
  obj@ptr <- NULL  # You can initialize the ptr slot as needed
  return(obj)
}

## Add ptr slot
gs_with_ptr <- SpatRasterWithPtr(gs)
gs_with_ptr@ptr <- gs@cpp

# Test again
try(gs_with_ptr@ptr$res[1] != gs_with_ptr@ptr$res[2])
#> [1] TRUE

try(gs_with_ptr@cpp$res[1] != gs_with_ptr@cpp$res[2])
#> [1] TRUE

try(res(gs_with_ptr)[1] != res(gs_with_ptr)[2])
#> [1] TRUE
```

<sup>Created on 2024-01-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by wlangera** [comment]
This is apparently already fixed in #10

**Comment by STBrinkmann** [comment]
Hi,
first of all let me apologize for my long absence. I have moved the GVI package (and others) to a new R package [CGEI](https://github.com/STBrinkmann/CGEI). This package will provide tools used in spatial epidemiology and urban planning.

The issue you have mentioned is fixed in CGEI. Also, I have improved and added functionalists to the existing GVI functions.
